### PR TITLE
Show help only for the error itself has to do with a command, argument, or option being invalid with dotnet nuget command

### LINKED ISSUE #10788: Show subcommand help instead of main dotnet nuget trust command help for all cases
## Details about Problem

NuGet product used (dotnet.exe):

Product version: 

Worked before? If so, with which NuGet version: N/A

## Repro steps and/or sample project
[Printing out help doesn't help at all unless the error itself has to do with a command, argument, or option being invalid.](https://github.com/NuGet/NuGet.Client/pull/3989/files#r617113765)
Current experience show help for `dotnet nuget` is different from `nuget.exe`. It show help  when there is a exception like InvalidOperationException even though it's nothing to do with `a command, argument, or option being invalid`

For example:

It adds red marked help even though it's not really needed.

### PR DESCRIPTION
**Pull request by erdembayar**
## Bug

Fixes: https://github.com/NuGet/Home/issues/10788

Regression? Last working version:

### Implemented sub commands so it show subcommand help instead of main `dotnet nuget trust` command help for all cases.
Previously, all options some not related to sub-command is displayed:
image

After, only relevant info for subcommand displayed.
image

### Even main command help is  changed:
Previously, doesn't show sub-commands.
image

After, sub commands are listed below:
image

### Duplicate entry error(when try to add same trust certificate multiple times), doesn't display help info anymore:
Before:
image
After:
image

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [x] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

File: src/NuGet.Core/NuGet.Commands/StringFormatter.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Globalization;
+
+namespace NuGet.Commands
+{
+    public static class StringFormatter
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Commands/StringFormatter.cs]: Does this need to be public?

I'd argue the class name is not descriptive enough to warrant that.

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Strings.resx [hunk 2]
```diff
@@ -852,4 +852,46 @@ The search is a case-insensitive string comparison using the supplied value, whi
   <data name="SignCommandNoCertificateException" xml:space="preserve">
     <value>No certificate was provided.</value>
   </data>
+  <data name="TrustAuthorCommandDescription" xml:space="preserve">
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Strings.resx:855]: Have these strings been reviewed by a PM?

@chgill-MSFT

File: src/NuGet.Core/NuGet.Commands/StringFormatter.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    {
+        public static string Log_TrustedSignerAlreadyExists(
+            string packageId)
+        {
+            return string.Format(CultureInfo.CurrentCulture,
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Commands/StringFormatter.cs]: I would just duplicate/move the strings instead of exposing a helper in NuGet.Commands.

File: src/NuGet.Core/NuGet.Commands/StringFormatter.cs (hunk 1 above)
Comment (erdembayar) [reply src/NuGet.Core/NuGet.Commands/StringFormatter.cs]: Ok. Removed.

File: src/NuGet.Core/NuGet.Commands/StringFormatter.cs (hunk 3 above)
Comment (erdembayar) [reply src/NuGet.Core/NuGet.Commands/StringFormatter.cs]: Sure, I did it.

**Comment by kartheekp-ms** [comment]
Erick - I had a quick look at the code, looks good so far. I think we need a follow-up issue to register subcommands in `dotnet/sdk` repo also. https://github.com/dotnet/sdk/pull/18061#discussion_r645194582

**Comment by erdembayar** [comment]
> Erick - I had a quick look at the code, looks good so far. I think we need a follow-up issue to register subcommands in `dotnet/sdk` repo also. [dotnet/sdk#18061 (comment)](https://github.com/dotnet/sdk/pull/18061#discussion_r645194582)

Sure, created follow up issue. https://github.com/dotnet/sdk/issues/20249

File: src/NuGet.Clients/NuGet.PackageManagement.UI/NuGet.PackageManagement.UI.csproj [hunk 4]
```diff
@@ -542,4 +542,4 @@
     </Content>
   </ItemGroup>
   <ItemGroup />
-</Project>
+</Project>
```
Comment (nkolev92) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/NuGet.PackageManagement.UI.csproj]: Unnecessary change.

**Review by nkolev92 — COMMENTED** [review]
I'm happy to approve, I just want to make sure the user facing strings have been validated.

**Comment by erdembayar** [comment]
@chgill-MSFT 
Please validate newly added strings here.

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Strings.resx [hunk 5]
```diff
@@ -852,4 +852,57 @@ The search is a case-insensitive string comparison using the supplied value, whi
...
+    <value>The name of the existing trusted signer to sync.</value>
+  </data>
+  <data name="TrustedSignerNameToAdd" xml:space="preserve">
+    <value>The name of the trusted signer to add. If NAME already exists in the configuration, the signature is appended.</value>
+  </data>
+  <data name="TrustedSignerNameToRemove" xml:space="preserve">
+    <value>The name of the existing trusted signer to remove.</value>
+  </data>
+  <data name="TrustLocalSignedNupkgPath" xml:space="preserve">
+    <value>The given PACKAGE should be a local path to the signed .nupkg file.</value>
```
Comment (chrisraygill) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Strings.resx]: Should PACKAGE here be in all caps?

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Strings.resx [hunk 6]
```diff
@@ -852,4 +852,57 @@ The search is a case-insensitive string comparison using the supplied value, whi
...
+    <value>The fingerprint of the certificate.</value>
+  </data>
+  <data name="TrustedCertificateSignerNameToAdd" xml:space="preserve">
+    <value>The name of the trusted signer to add. If a trusted signer with the given name already exists, the certificate item is added to that signer. Otherwise a trusted author is created with a certificate item from the given certificate information.</value>
+  </data>
+  <data name="TrustedSignerNameExists" xml:space="preserve">
+    <value>The name of the existing trusted signer to sync.</value>
+  </data>
+  <data name="TrustedSignerNameToAdd" xml:space="preserve">
+    <value>The name of the trusted signer to add. If NAME already exists in the configuration, the signature is appended.</value>
```
Comment (chrisraygill) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Strings.resx]: Should NAME be in all caps?

File: src/NuGet.Core/NuGet.Commands/Strings.resx [hunk 7]
```diff
@@ -1131,4 +1131,7 @@ For more information, visit https://docs.nuget.org/docs/reference/command-line-r
     <value>'{0}' was resolved as a dependency of '{1}', but the dependency is using 'Xamarin.iOS' while '{1}' is using '{2}' as its TargetFramework. There might be compatibility issues when MacCatalyst projects depend on Xamarin.iOS projects.</value>
     <comment>0 - library/package name, 1 - VS project name, 2 - target framework</comment>
   </data>
-</root>
+  <data name="Error_TrustFingerPrintAlreadyExist" xml:space="preserve">
+    <value>The certificate finger you're trying to add is already in certificate fingerprint list.</value>
```
Comment (chrisraygill) [inline src/NuGet.Core/NuGet.Commands/Strings.resx]:
```suggestion
    <value>The certificate finger you're trying to add is already in the certificate fingerprint list.</value>
```

**Review by chrisraygill — APPROVED** [review]
Overall LGTM, a couple minor questions and one word suggestion.

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Strings.resx (hunk 6 above)
Comment (erdembayar) [reply src/NuGet.Core/NuGet.CommandLine.XPlat/Strings.resx]: I changed to non-capital.

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Strings.resx (hunk 5 above)
Comment (erdembayar) [reply src/NuGet.Core/NuGet.CommandLine.XPlat/Strings.resx]: I changed to non-capital.

**Comment by erdembayar** [comment]
> I'm happy to approve, I just want to make sure the user facing strings have been validated.

I just addressed comments by Chris.

**Comment by erdembayar** [comment]
@nkolev92 
I have approval from Chris. Please review again.

**Comment by erdembayar** [comment]
@kartheekp-ms 
Please review.

**Comment by erdembayar** [comment]
@kartheekp-ms @heng-liu 
Please review.

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs [hunk 8]
```diff
@@ -23,107 +22,321 @@ internal static class TrustedSignersCommand
...
+                        "--configfile",
+                        Strings.Option_ConfigFile,
+                        CommandOptionType.SingleValue);
+
+                    syncCommand.HelpOption(XPlatUtility.HelpOption);
+
+                    CommandOption verbosity = syncCommand.Option(
+                        "-v|--verbosity",
+                        Strings.Verbosity_Description,
+                        CommandOptionType.SingleValue);
```
Comment (kartheekp-ms) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs]: Given that these options are common for all or majority of subcommands, I think we should abstract this logic for better code reuse just like `HelpOption`.

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs [hunk 9]
```diff
@@ -23,107 +22,321 @@ internal static class TrustedSignersCommand
...
+
+                    CommandArgument name = syncCommand.Argument("<NAME>",
+                                               Strings.TrustedSignerNameExists);
```
Comment (kartheekp-ms) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs:57]: does this command accept `multiple values`? If yes, we should pass `multipleValues` parameter to `true`. This comment applies for all subcommand's `arguments`.

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs [hunk 10]
```diff
@@ -23,107 +22,321 @@ internal static class TrustedSignersCommand
...
-                    string fingerprint = null;
-                    if (command.Values.Count() > 2)
+                var runner = new TrustedSignersCommandRunner(trustedSignersProvider, sourceProvider);
+                Task<int> trustedSignTask = runner.ExecuteCommandAsync(trustedSignersArgs);
+                return await trustedSignTask;
+            }
+            catch (Exception e)
+            {
+                // nuget trust command handled exceptions.
+                if (e is InvalidOperationException)
```
Comment (kartheekp-ms) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs]: I think we should handle `InvalidOperationException` and `ArgumentException` in seperate `catch` blocks instead of handling all the exceptions in the generic exception handler. 

How about making subcommands responsible for throwing appropriate exceptions and here we just `log the error` and return `exit code: 1`?

**Review by kartheekp-ms — COMMENTED** [review]
I have reviewed product code but not tests yet.

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs (hunk 10 above)
Comment (erdembayar) [reply src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs]: I separated the try catch blocks.
Not sure subcommands throwing appropriate exception, it's just rethrowing exception, much more code but end result is same.

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs (hunk 8 above)
Comment (erdembayar) [reply src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs]: I created `AddVerbosity` extension method, now syntax is shorter and clean.

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs (hunk 9 above)
Comment (erdembayar) [reply src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs:57]: No, it's not. Here only `owners`  has multivalues and already have correct parameter value.

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Strings.resx (hunk 2 above)
Comment (erdembayar) [reply src/NuGet.Core/NuGet.CommandLine.XPlat/Strings.resx:855]: Chris already approved it. Could you be able to review again?

Comment (nkolev92) [reply src/NuGet.Core/NuGet.CommandLine.XPlat/Strings.resx:855]: I don't really have any other concerns here.
Not signing off, but you already have the necessary 1 approval.

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs [hunk 11]
```diff
@@ -23,107 +22,305 @@ internal static class TrustedSignersCommand
...
+                        CommandOptionType.SingleValue);
+
+                    authorCommand.HelpOption(XPlatUtility.HelpOption);
 
-                CommandOption configFile = trustedSignersCmd.Option(
+                    CommandOption verbosity = authorCommand.VerbosityOption();
+
+                    CommandArgument name = authorCommand.Argument("<NAME>",
+                                               Strings.TrustedSignerNameToAdd);
+                    CommandArgument package = authorCommand.Argument("<PACKAGE>",
```
Comment (kartheekp-ms) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs:106]: As per [spec](https://github.com/NuGet/Home/blob/b30a8ff077272045224829c6772327b724ebed35/proposed/2021/DotnetNugetTrustedSigners.md#note), `For trust author and trust repository, <package> may be a glob/wildcard. This must be consistent across platforms, so the Windows version must expand these wildcards when it receives them as arguments.`. There was a similar requirement for `dotnet nuget verify` command, hence we passed `multipleValues: true`.

https://github.com/NuGet/NuGet.Client/blob/966b1b21eec28834365a8604744dab682c8326ea/src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/VerifyCommand.cs#L24-L27

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs [hunk 12]
```diff
@@ -23,107 +22,305 @@ internal static class TrustedSignersCommand
...
+                                               Strings.TrustLocalSignedNupkgPath);
+
+                    authorCommand.OnExecute(async () =>
+                    {
+                        return await ExecuteCommand(TrustCommand.Author, algorithm: null, allowUntrustedRootOption.HasValue(), owners: null, verbosity, configFile, getLogger, setLogLevel, name: name.Value, sourceUrl: null, packagePath: package.Value);
```
Comment (kartheekp-ms) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs:111]: If we change `packagePath` to accept multiple values then we have to pass `packagePath.Values`.

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs [hunk 13]
```diff
@@ -23,107 +22,305 @@ internal static class TrustedSignersCommand
...
+                        "--owners",
+                        Strings.TrustCommandOwners,
+                        CommandOptionType.MultipleValue);
+
+                    CommandOption verbosity = repositoryCommand.VerbosityOption();
+
+                    CommandArgument name = repositoryCommand.Argument("<NAME>",
+                                               Strings.TrustedSignerNameToAdd);
+                    CommandArgument package = repositoryCommand.Argument("<PACKAGE>",
+                                               Strings.TrustLocalSignedNupkgPath);
```
Comment (kartheekp-ms) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs:141]: As per [spec](https://github.com/NuGet/Home/blob/b30a8ff077272045224829c6772327b724ebed35/proposed/2021/DotnetNugetTrustedSigners.md#note), `For trust author and trust repository, <package> may be a glob/wildcard. This must be consistent across platforms, so the Windows version must expand these wildcards when it receives them as arguments.`. There was a similar requirement for `dotnet nuget verify` command, hence we passed `multipleValues: true`.

https://github.com/NuGet/NuGet.Client/blob/966b1b21eec28834365a8604744dab682c8326ea/src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/VerifyCommand.cs#L24-L27

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs [hunk 14]
```diff
@@ -23,107 +22,305 @@ internal static class TrustedSignersCommand
...
+
+                    repositoryCommand.OnExecute(async () =>
+                    {
+                        return await ExecuteCommand(TrustCommand.Repository, algorithm: null, allowUntrustedRootOption.HasValue(), owners: owners, verbosity, configFile, getLogger, setLogLevel, name: name.Value, sourceUrl: null, packagePath: package.Value);
```
Comment (kartheekp-ms) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs:145]: If we change `packagePath` to accept multiple values, then we have to pass `packagePath.Values`.

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs [hunk 15]
```diff
@@ -23,107 +22,305 @@ internal static class TrustedSignersCommand
...
+                        "--configfile",
+                        Strings.Option_ConfigFile,
+                        CommandOptionType.SingleValue);
+
+                    sourceCommand.HelpOption(XPlatUtility.HelpOption);
+
+                    CommandOption owners = sourceCommand.Option(
+                        "--owners",
+                        Strings.TrustCommandOwners,
+                        CommandOptionType.MultipleValue);
```
Comment (kartheekp-ms) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs]: As per [doc](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-nuget-trust#options-5), `--owners` option accepts list of values separated by semi-colon. 

Copied from examples section in the doc, `dotnet nuget trust source NuGetTrust https://api.nuget.org/v3/index.json --owners "Nuget;Microsoft"`

I think `CommandOptionType` for this option should be `SingleValue` instead of `MultipleValue`. This comment applies to all the sub-commands which has `owners` as an option.

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs (hunk 11 above)
Comment (erdembayar) [reply src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs:106]: We had discussion on this one and decided not to in favor in security concern. https://github.com/NuGet/Home/issues/10647#issuecomment-801523491

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs (hunk 12 above)
Comment (erdembayar) [reply src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs:111]: See https://github.com/NuGet/NuGet.Client/pull/4211#discussion_r720614519

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs (hunk 11 above)
Comment (erdembayar) [reply src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs:106]: `Verify` command doesn't mutate anything so no security concern.

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs (hunk 13 above)
Comment (erdembayar) [reply src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs:141]: See https://github.com/NuGet/NuGet.Client/pull/4211#discussion_r720614519

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs (hunk 15 above)
Comment (erdembayar) [reply src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs]: Ok. I addressed your concerns. Please review again.

**Comment by erdembayar** [comment]
@kartheekp-ms 
If you don't have any other comment then I'll merge today evening. Nikolche already tacitly approved.

**Comment by erdembayar** [comment]
> @kartheekp-ms If you don't have any other comment then I'll merge today evening. Nikolche already tacitly approved.

Actually I need another approval to merge. Please review.

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs [hunk 16]
```diff
@@ -23,107 +22,305 @@ internal static class TrustedSignersCommand
...
+                setLogLevel(XPlatUtility.MSBuildVerbosityToNuGetLogLevel(verbosity.Value()));
+
+#pragma warning disable CS0618 // Type or member is obsolete
+                var sourceProvider = new PackageSourceProvider(settings, enablePackageSourcesChangedEvent: false);
+#pragma warning restore CS0618 // Type or member is obsolete
+                var trustedSignersProvider = new TrustedSignersProvider(settings);
+
+                var runner = new TrustedSignersCommandRunner(trustedSignersProvider, sourceProvider);
+                Task<int> trustedSignTask = runner.ExecuteCommandAsync(trustedSignersArgs);
+                return await trustedSignTask;
```
Comment (kartheekp-ms) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs:277]:
``` suggestion
                return trustedSignTask;
```

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs [hunk 17]
```diff
@@ -23,107 +22,305 @@ internal static class TrustedSignersCommand
...
-                    CommandOptionType.MultipleValue);
+                    CommandOption verbosity = listCommand.VerbosityOption();
 
-                CommandOption verbosity = trustedSignersCmd.Option(
-                    "-v|--verbosity",
-                    Strings.Verbosity_Description,
-                    CommandOptionType.SingleValue);
+                    listCommand.OnExecute(async () =>
+                    {
+                        return await ExecuteCommand(TrustCommand.List, algorithm: null, allowUntrustedRootOption: false, owners: null, verbosity, configFile, getLogger, setLogLevel);
```
Comment (kartheekp-ms) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs:40]:
``` suggestion
                        return ExecuteCommand(TrustCommand.List, algorithm: null, allowUntrustedRootOption: false, owners: null, verbosity, configFile, getLogger, setLogLevel);

```

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs [hunk 18]
```diff
@@ -23,107 +22,305 @@ internal static class TrustedSignersCommand
...
+#pragma warning disable CS0618 // Type or member is obsolete
+                var sourceProvider = new PackageSourceProvider(settings, enablePackageSourcesChangedEvent: false);
+#pragma warning restore CS0618 // Type or member is obsolete
+                var trustedSignersProvider = new TrustedSignersProvider(settings);
+
+                var runner = new TrustedSignersCommandRunner(trustedSignersProvider, sourceProvider);
+                Task<int> trustedSignTask = runner.ExecuteCommandAsync(trustedSignersArgs);
+                return await trustedSignTask;
+            }
+            catch (InvalidOperationException e)
```
Comment (kartheekp-ms) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs:279]: More I think about it, the correct place for exception handling would have been within in the `subcommand.OnExecute` code block because the handlers are specific to the sub-commands. It is possible that I may be missing something here.

**Review by kartheekp-ms — APPROVED** [review]

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs (hunk 16 above)
Comment (erdembayar) [reply src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs:277]: Even though this suggestion saves 1 machine state, but makes hard to track down aggregated exception(I couldn't find the link for ref, it brought up during last year learning day for async programming), maybe better to keep as it's since all other `dotnet nuget` implementation doing same.

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs (hunk 17 above)
Comment (erdembayar) [reply src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs:40]: See https://github.com/NuGet/NuGet.Client/pull/4211#discussion_r722577744

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs (hunk 18 above)
Comment (erdembayar) [reply src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs:279]: I'll address this one next time I do https://github.com/NuGet/Home/issues/10294
